cmd/secretless-broker/main.go

```diff
@@ -41,5 +77,68 @@ func main() {
 	// Either the flag or the arg should be enough to show the version
 	params.ShowVersion = *showVersion || flag.Arg(0) == "version"
 
-	secretless.Start(&params, plugin.GetManager())
+	return params
+}
+
+func NewKillSignalChan() chan os.Signal {
+	signalChannel := make(chan os.Signal, 1)
+	signal.Notify(signalChannel,
+		syscall.SIGABRT,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGQUIT,
+		syscall.SIGTERM,
+	)
+	return signalChannel
+}
+
+type Stopper interface {
+	Stop()
+}
+
+// StopOnKillSignal will make any "Stopper" automatically Stop() when OS kill
+// signals are received.
+// TODO: Possible synchronization to ensure all go routines complete.  Also note,
+// the usage shown above in main relies on the fact that these will be triggered
+// in the order they're setup -- an assumption that relies on impl details of
+// the signal package.  Not ideal.
+func StopOnKillSignal(s Stopper) {
+	killSignals := NewKillSignalChan()
+
+	go func() {
+		<-killSignals
+		s.Stop()
+	}()
+}
+
+// type is "memory" or "cpu"
+type PerformanceProfile interface {
+	Start()
+	Stop()
+}
```

Comment:
If both PerformanceProfile and Secretless implement the same methods interface, it might be good to find a common name and extract into a separate package that both of them inherit

Reply:
> that both of them inherit
Important nitpick: They don't inherit.  They just implement :)

But essentially what you're getting at is that they're both just "services" that can be started and stopped.  Perhaps the interface name should just be `Service`?  

You could call it a `StartStopper` in analogy with `ReadWriter`?  I kind of hate that but it's Go-ish...

Reply:
> But essentially what you're getting at is that they're both just "services" that can be started and stopped. Perhaps the interface name should just be Service

yup!